addons/pos_discount/static/src/js/DiscountButton.js

```diff
@@ -35,6 +35,8 @@ odoo.define('pos_discount.DiscountButton', function(require) {
                 });
                 return;
             }
+            // Remove discount taxes
+            product.taxes_id = [];
```

Comment:
Let's not do this. Let the user put tax on their discount product.

Reply:
@caburj Thanks for your review. 
It causes some issues if it contains a tax. Could you please explain me why they need to choose a tax for the discount product?

Reply:
I received comments just few days after I returned to pos team, that there are pos users who purposely put tax to their discount products. So unless we're very sure about that override (remove directly the tax specified by the user), then we should not do it.

Reply:
@caburj I removed it.